### PR TITLE
Fix detected corner cases

### PR DESCRIPTION
**Pull request by uurien**
### What does this PR do?

Fix some detected taint tracking corner cases

### Checklist

- [ ] The CHANGELOG.md has been updated
- [x] Unit tests have been updated and pass
- [ ] If known, an appropriate milestone has been selected

### CODE REVIEW

File: src/api/string_methods.cc [hunk 1]
```diff
@@ -65,8 +65,10 @@ void NewTaintedString(const FunctionCallbackInfo<Value>& args) {
     auto parameterName = args[2];
     auto type = args[3];
 
-    // if string length == 1 then make a new one in order to avoid chache issues.
+    // if string length < 10 then make a new one in order to avoid cache issues.
     if (v8::Local<v8::String>::Cast(args[1])->Length() == 1) {
+        parameterValue = tainted::NewExternalString(isolate, args[1]);
+    } else if (v8::Local<v8::String>::Cast(args[1])->Length() < 10) {
```
Comment (hoolioh) [inline src/api/string_methods.cc:71]:
```suggestion 
    // if string length < 10 then make a new one in order to avoid cache issues.
    int len =  v8::Local<v8::String>::Cast(args[1])->Length();
    if (len == 1) {
        parameterValue = tainted::NewExternalString(isolate, args[1]);
    } else if (len < 10) {
```

**Review by hoolioh — APPROVED** [review]
